src/nvim/ex_docmd.c

```diff
@@ -8544,7 +8544,14 @@ eval_vars (
       if (autocmd_fname != NULL && !path_is_absolute(autocmd_fname)) {
         // Still need to turn the fname into a full path.  It was
         // postponed to avoid a delay when <afile> is not used.
-        result = (char_u *)FullName_save((char *)autocmd_fname, false);
+        if (has_event(EVENT_CMDLINEENTER)
+            || has_event(EVENT_CMDLINELEAVE)
+            || has_event(EVENT_CMDWINENTER)
+            || has_event(EVENT_CMDWINLEAVE)) {
+          result = (char_u *)tv_dict_get_string(get_vim_var_dict(VV_EVENT), "cmdtype", true);
```

Comment:
Doesn't `result = xstrdup(autocmd_fname)` work? All these four autocommands should have set the same value to `fname`.

Reply:
Also this if block changes unconditionally whenever any such autocommand is _defined_, not whether the actual _invoked_ event was one of these for. It doesn't make sense to me.

Reply:
@bfredl How can I check the currently invoked event?